docs/spec/distributing.rst

```diff
@@ -10,72 +10,154 @@ Stub files
+* The ellipsis literal, ``...``, which can stand in for any value
+* Any value that is a
+  :ref:`legal parameter for typing.Literal <literal-legal-parameters>`
+* Floating point literals, such as ``3.14``
+* Complex literals, such as ``1 + 2j``
 
-* Modules that use annotations for other purposes
+Module Level Attributes
+"""""""""""""""""""""""
 
-Stub files have the same syntax as regular Python modules.  There is one
-feature of the ``typing`` module that is different in stub files:
-the ``@overload`` decorator described below.
+Type checkers should support module-level variable annotations, with and without
+assignments::
 
-The type checker should only check function signatures in stub files;
-It is recommended that function bodies in stub files just be a single
-ellipsis (``...``).
+    x: int
+    x: int = 0
+    x = 0  # type: int
+    x = ...  # type: int
 
-The type checker should have a configurable search path for stub files.
-If a stub file is found the type checker should not read the
-corresponding "real" module.
+The :ref:`Literal shortcut using Final <literal-final-interactions>` should be
+supported::
 
-While stub files are syntactically valid Python modules, they use the
-``.pyi`` extension to make it possible to maintain stub files in the
-same directory as the corresponding real module.  This also reinforces
-the notion that no runtime behavior should be expected of stub files.
+    x: Final = 0  # type is Literal[0]
 
-Additional notes on stub files:
+When the type of a variable is omitted or disagrees from the assigned value,
+type checker behavior is undefined::
 
-* Modules and variables imported into the stub are not considered
-  exported from the stub unless the import uses the ``import ... as
-  ...`` form or the equivalent ``from ... import ... as ...`` form.
-  (*UPDATE:* To clarify, the intention here is that only names
-  imported using the form ``X as X`` will be exported, i.e. the name
-  before and after ``as`` must be the same.)
+    x = 0  # behavior undefined
+    x: Final = ...  # behavior undefined
+    x: int = ""  # behavior undefined
 
-* However, as an exception to the previous bullet, all objects
-  imported into a stub using ``from ... import *`` are considered
-  exported.  (This makes it easier to re-export all objects from a
-  given module that may vary by Python version.)
+Classes
+"""""""
 
-* Just like in `normal Python files <https://docs.python.org/3/reference/import.html#submodules>`_, submodules
-  automatically become exported attributes of their parent module
-  when imported. For example, if the ``spam`` package has the
-  following directory structure::
+Class definition syntax follows general Python syntax, but type checkers
+are expected to understand only the following constructs in class bodies:
 
-      spam/
-          __init__.pyi
-          ham.pyi
+* The ellipsis literal ``...`` is used for empty class bodies. Using ``pass`` in
```

Comment:
That seems unnecessarily strict; it seems fine that `pass` is also allowed (even if it's not stylistically preferred).

Reply:
I think it's OK to support `pass`, but we may need to revisit this question if/when we standardize the way to mark a protocol method as abstract versus implemented. In source files, type checkers use (non-specified) heuristics to determine whether the implementation is "trivial", but different rules are needed for stubs.

Reply:
Again, I don't feel super strongly about this, but I slightly prefer to not allow `pass` because allowing it would mean that type checkers need to support two ways of expressing the same thing, for no clear benefit.

Reply:
I don't think it's appropriate for the typing spec to say whether `pass` should be "disallowed" in a stub. After all, type checkers don't report errors in stubs that are imported by programs.

The typing spec should indicate whether a type checker should be required to handle `pass` in a stub. I'm ambivalent on this question. If we don't say that it's required, then the behavior will be undefined.

For what it's worth, I have seen `pass` used in some stubs. Not in typeshed, but in other stubs.

Reply:
I don't think anyone's advocating that we require `pass` to be an error. As I understand them, the options being considered are (1) specify that `pass` in class and function bodies means the same thing as `...` (the "allow" option),or (2) leave the behavior of `pass` undefined (the "disallow" option).

Reply:
Thinking about this again, the current text is fine.